### PR TITLE
skip 404.html pages in sitemap generation

### PR DESCRIPTION
**Pull request by manigandham**
## Changes
- Sitemap generation skips `404.html` page (used for routes not otherwise found for the site).

## Testing
- Updated sitemap test fixture (`astro-rss`) to include a `404.astro` page that should not be included in the sitemap output.

## Docs
- Nothing to update in docs

### CODE REVIEW

File: packages/astro/src/build/sitemap.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   const uniqueURLs = new Set<string>();
 
   // TODO: find way to respect <link rel="canonical"> URLs here
-  // TODO: find way to exclude pages from sitemap
+  // TODO: find way to exclude pages from sitemap (currently only skips 404 pages)
 
   // look through built pages, only add HTML
   for (const id of Object.keys(buildState)) {
     if (buildState[id].contentType !== 'text/html') continue;
+    if (id.startsWith('404.html')) continue;
```
Comment (FredKSchott) [inline packages/astro/src/build/sitemap.ts:14]: I think `id === "404.html"` makes a big more sense, since html is the end of the id

**Review by FredKSchott — COMMENTED** [review]
LGTM once you can get tests passing

File: packages/astro/src/build/sitemap.ts (hunk 1 above)
Comment (manigandham) [reply packages/astro/src/build/sitemap.ts]: Done.

**Comment by manigandham** [comment]
@FredKSchott All set, The file paths contain the starting slash so the correct check is for `/404.html`

**Comment by FredKSchott** [comment]
Thanks @manigandham!

**Comment by manigandham** [comment]
Forgot to add the changeset to this PR so it looks like it didn't get added to the release... should I open a new one with the same commits?

**Comment by FredKSchott** [comment]
oh dang, we should have caught that as well. If you can create a new PR with just the changeset (generated by running `yarn changeset` in the top-level project) we can get that merged seperately. Thanks!

**Pull request merged**